SpatialGDK/Source/Private/Interop/SnapshotManager.cpp

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SnapshotManager.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "Interop/Connection/SpatialWorkerConnection.h"
+#include "Interop/GlobalStateManager.h"
+#include "Interop/SpatialReceiver.h"
+#include "SpatialConstants.h"
+#include "Utils/SchemaUtils.h"
+
+DEFINE_LOG_CATEGORY(LogSnapshotManager);
+
+using namespace improbable;
+
+void USnapshotManager::Init(USpatialNetDriver* InNetDriver, UGlobalStateManager* InGlobalStateManager)
+{
+	NetDriver = InNetDriver;
+	Receiver = InNetDriver->Receiver;
+	GlobalStateManager = InGlobalStateManager;
+}
+
+// WorldWipe will send out an expensive entity query for every entity in the deployment.
+// It does this by having an entity query for all entities that are not the GSM (workaround for not having the ability to make a query for all entities).
+// Once it has the response to this query, it will send deletion requests for all found entities and then one for the GSM itself.
+// Should only be triggered by the worker which is authoritative over the GSM.
+void USnapshotManager::WorldWipe(const USpatialNetDriver::ServerTravelDelegate& Delegate)
+{
+	UE_LOG(LogSnapshotManager, Warning, TEXT("World wipe for deployment has been triggered. All entities will be deleted."));
+
+	Worker_Constraint GSMConstraint;
+	GSMConstraint.constraint_type = WORKER_CONSTRAINT_TYPE_ENTITY_ID;
```

Comment:
What do you think about having some helper which constructs common queries?

Reply:
Discussed this. The queries are actually quite different and only being used in one place each so it doesn't make sense to turn them into common code.